app/src/main/res/layout/http_authentication.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~ Copyright (c) 2019 DuckDuckGo
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:paddingTop="20dp">
+
+
+    <TextView
+        android:id="@+id/httpAuthInformationText"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:layout_marginStart="20dp"
+        android:layout_marginEnd="20dp"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <com.google.android.material.textfield.TextInputLayout
+        android:id="@+id/httpAuthUsernameContainer"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="20dp"
+        android:layout_marginEnd="20dp"
+        android:layout_marginTop="20dp"
+        app:layout_constraintTop_toBottomOf="@id/httpAuthInformationText">
+
+        <EditText
```

Comment:
We might want to mark these fields with autocomplete hints, for example, for the `username` one:

     android:importantForAutofill="yes"
     android:autofillHints="username"
     tools:ignore="UnusedAttribute" 

And `android:autofillHints="password"` for the password field.

That said, I haven't tested this, and we haven't done it anywhere else in the app before. So leaving a note here as a future note - feel free to ignore this for now.

Reply:
When testing I was able to use 1Password to fill in the form so I wasn't sure if these were needed. Happy to add for further compatibly.

Reply:
If it's working for you as is, feel free to leave it. It can be a small addition later if it looks like it would be needed.

As I understand it, autofill heuristics will automatically determine fields' eligibility for autofill if it can. So that's probably why we're getting it for free. The EditText is probably inspected and it likely knows it's a username and password.

We can add these explicitly later in a follow up if needed.